polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -2417,6 +2534,29 @@ impl<T: Config> Pallet<T> {
 					}
 				}
 			}
+			stage = RemoveExpiredAliasAuthorizations;
+		}
+		if stage == RemoveExpiredAliasAuthorizations {
+			let block_num = frame_system::Pallet::<T>::block_number().saturated_into::<u64>();
+			let mut writes = 0;
+			// need to iterate keys and modify map in separate steps to avoid undefined behavior
+			let keys: Vec<VersionedLocation> = AuthorizedAliases::<T>::iter_keys().collect();
```

Comment:
this might be too many reads, we should check the `weight_cutoff` on every iteration

Reply:
do we need to in practice? iterating over even a million keys should fit in practice, no?

I'm not even sure how we could do partial iterations, how to resume from where we left off...

Reply:
in the context of this function (may be I missing a broader picture), it looks wrong. a client passes `weight_cutoff`, for this stage we ignore it. we do not know how small or big the `weight_cutoff`.

I think we can iterate one by one and if there was no enough weight to complete, we can return `RemoveExpiredAliasAuthorizations(Some(cursor))`.